pkgs/development/python-modules/markitdown/default.nix

```diff
@@ -0,0 +1,75 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  beautifulsoup4,
+  ffmpeg-headless,
+  mammoth,
+  markdownify,
+  numpy,
+  openpyxl,
+  pandas,
+  pathvalidate,
+  pdfminer-six,
+  puremagic,
+  pydub,
+  python-pptx,
+  requests,
+  speechrecognition,
+  youtube-transcript-api,
+  pytestCheckHook,
+}:
+
+buildPythonPackage {
+  pname = "markitdown";
+  version = "unstable-2024-12-15";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "markitdown";
+    rev = "6dad1cca967f0d8eb15233104ab5b30fc37e6649";
+    hash = "sha256-oview6cA4bFPq5fp6UegW34uKHF1PybAh+FUcfgPa+Q=";
+  };
+
+  build-system = [
+    hatchling
+  ];
+
+  dependencies = [
+    beautifulsoup4
+    ffmpeg-headless
+    mammoth
+    markdownify
+    numpy
+    openpyxl
+    pandas
+    pathvalidate
+    pdfminer-six
+    puremagic
+    pydub
+    python-pptx
+    requests
+    speechrecognition
+    youtube-transcript-api
+  ];
+
+  pythonImportsCheck = [
+    "markitdown"
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  disabledTests = [
+    # Require network access
+    "test_markitdown_remote"
```

Comment:
It also checks for the `GITHUB_ACTIONS` environment variable to skip [1]. This is also good tho.

[1] https://github.com/microsoft/markitdown/blob/19c111251b779bbe10ff319e59a7899545cd4b64/tests/test_markitdown.py#L11